src/serac/numerics/functional/detail/hexahedron_H1.inl

```diff
@@ -144,30 +147,36 @@ struct finite_element<mfem::Geometry::CUBE, H1<p, c> > {
     using flux_t   = decltype(get<0>(get<1>(in_t{})) + dot(get<1>(get<1>(in_t{})), tensor<double, dim>{}));
 
     tensor<tuple<source_t, flux_t>, q * q * q> output;
+    auto                                       x_range = RAJA::RangeSegment(0, q * q * q);
+    RAJA::loop<threads_x>(ctx, x_range, [&](int Q) {
+      int qx = Q % q;
+      int qy = (Q / q) % q;
+      int qz = (Q / (q * q));

```

Comment:
Wouldn't it be better and more flexible to use a [RAJA::kernel](https://raja.readthedocs.io/en/develop/sphinx/user_guide/feature/loop_basic.html#complex-loops-raja-kernel) to handle this multi-level for-loop.